walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
+
+  private final int TOTAL_DAY_MINUTES = 60 * 24;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> availableTimes;
+
+    // Declaration of an array of boolean that indicates the available minutes in a day.
+    Boolean[] availableMinutesAttendees = new Boolean[TOTAL_DAY_MINUTES];
+    Arrays.fill(availableMinutesAttendees, true);
+
+    Boolean[] availableMinutesOptAttendees = new Boolean[TOTAL_DAY_MINUTES];
+    Arrays.fill(availableMinutesOptAttendees, true);
+
+    Collection<String> attendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
```

Comment:
```suggestion
    Collection<String> optAttendees = request.getOptionalAttendees();
```
Nit: Good to use consistent language. Either "opt" or "optional" works, but make sure you choose one and stick to it : )

Reply:
Sure, thanks!